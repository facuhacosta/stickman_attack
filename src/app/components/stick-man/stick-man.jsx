import React, { useRef, useState } from 'react';
import style from './stick-man.module.scss';
import gif from '../../../public/assets/man.webm'
import { HealthBar } from '../health-bar';
  
export function StickMan({handleClick, propStyles, enemy_type, damage_dealed, position, animation_delay}) {
  const videoRef = useRef();
  const [health, setHealth] = useState(() => {
    switch (enemy_type) {
      case 'basic':
        return 100;
      case 'powered':
        return 200;
      case 'boss':
        return 400;
    }
  });
  const [maxHealth, setMaxHealth] = useState(health);

  const changeSpeed = () => {
    videoRef.current.playbackRate = 1;
  }

  const onClick = () => {
    // handleClick()
    setHealth(prev => prev - damage_dealed);
  }

  const barHealth = () => {
    return ((health/maxHealth)*100);
  }

  return (
    <div
      className={style['stick-man']}
      style={{
        bottom: `${position}%`,
        animationDelay: `${animation_delay}s`
      }}
    >
      <HealthBar completed={Math.round(barHealth())} health={health} />
      <video
        className={style['stick-man-video']}
        style={propStyles}
        ref={videoRef}
        onCanPlay={changeSpeed}
        onClick={() => onClick()}
        autoPlay
        muted
        loop
        data-testid="stickman-testId"
      >
        <source  src={gif} type="video/webm" />
      </video>
    </div>
  );
}