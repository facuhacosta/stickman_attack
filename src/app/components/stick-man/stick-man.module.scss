.stick-man {
  position: absolute;
  animation: move 5s linear;
  animation-fill-mode: forwards;
  width: 50px;
  visibility: hidden;
  &-video {
    width: 50px;

  }
}

@keyframes move {

  0% {
    transform: translateX(-100px);
    visibility: visible;
  }

  100% {
    transform: translateX(70vw);
    visibility: visible;
  }
}