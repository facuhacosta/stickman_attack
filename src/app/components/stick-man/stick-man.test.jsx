import React from 'react';
import { render, fireEvent, waitFor } from '@testing-library/react';
import { StickMan } from './stick-man';

describe('StickMan Component', () => {

  // Matching Snapshots
  test('it should match the snapshot', () => {
    const { asFragment } = render(<StickMan />);
    expect(asFragment()).toMatchSnapshot();
  });

  // Unit tests
  it('should log on click', () => {
    const consoleSpy = jest.spyOn(console, 'log');
    const { getByTestId } = render(<StickMan />);
    fireEvent.click(getByTestId('stickman-testId'));
    expect(consoleSpy).toHaveBeenCalledWith('click');
    jest.clearAllMocks();
  });

  // it('should log on click', async () => {
  //   const consoleSpy = jest.spyOn(console, 'log');
  //   const {  } = render(<StickMan />);
  //   await waitFor(() => {
  //     setTimeout(() => {
        
  //     }, 200);
  //     expect(consoleSpy).toHaveBeenCalledWith('change speed');
  //   })
  // });

  // it('should increment speed', () => {
  //   const changeSpeed  = jest.fn();
  //   const { getByTestId, debug } = render(<StickMan changeSpeed={changeSpeed} />);
  //   console.log(debug());
  //   waitFor(()=> {
  //     setTimeout(() => {
        
  //     }, 400);
  //   })
  //   expect(getByTestId('stickman-testId')).toHaveProperty('canplay');
  //   console.log(debug());
  //   expect(getByTestId('stickman-testId')).toHaveProperty('playbackrate', 2);
  //   expect(changeSpeed).toHaveBeenCalled();
    
  // });
});