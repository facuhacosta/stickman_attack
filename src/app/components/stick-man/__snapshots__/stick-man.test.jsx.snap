// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StickMan Component it should match the snapshot 1`] = `
<DocumentFragment>
  <video
    autoplay=""
    data-testid="stickman-testId"
    loop=""
  >
    <source
      src="[object Object]"
      type="video/webm"
    />
  </video>
</DocumentFragment>
`;
